### ISSUE TITLE
Default Dockerfile for ASP.Net Core Server Fails - base image not found

### ISSUE THREAD

**Issue by mws-github** [OpenAPITools/openapi-generator#19820]
# 🐛 Bug Report:

### Describe the bug
When trying to build the default generated Dockerfile on an ASP.Net Core generated project, I get the following error:

[+] Building 0.4s (3/3) FINISHED                                                                   docker:desktop-linux
 => [internal] load build definition from Dockerfile                                                               0.0s
 => => transferring dockerfile: 993B                                                                               0.0s
 => CANCELED [internal] load metadata for mcr.microsoft.com/dotnet/core/aspnet:8.0-buster-slim                     0.2s
 => ERROR [internal] load metadata for mcr.microsoft.com/dotnet/core/sdk:8.0-buster                                0.2s
------
 > [internal] load metadata for mcr.microsoft.com/dotnet/core/sdk:8.0-buster:
------
Dockerfile:10
--------------------
   8 |
   9 |     # Build container
  10 | >>> FROM mcr.microsoft.com/dotnet/core/sdk:8.0-buster AS build
  11 |
  12 |     # Copy the code into the container
--------------------
ERROR: failed to solve: mcr.microsoft.com/dotnet/core/sdk:8.0-buster: failed to resolve source metadata for mcr.microsoft.com/dotnet/core/sdk:8.0-buster: mcr.microsoft.com/dotnet/core/sdk:8.0-buster: not found

### Steps to Reproduce
Build a aspnetcore server, then try to build the docker file.

### Expected behavior
Expect the Dockerfile to build.

### Operation System (please complete the following information):
 - OS: Windows
 - Version: Latest

### Package System (please complete the following information):
 - Version: Latest ASP.Net Core Server Generator

### Additional Context ###
Looks like the base image is no longer valid (not found).

**Comment by wing328** [comment]
looks like an issue similar to https://github.com/microsoft/containerregistry/issues/144?

**Comment by MisinformedDNA** [comment]
Nope, .NET Core rebranded to .NET, so they removed /core from the name. Also, buster is Debian 10 and is considered EOL for .NET (only supporting up to .NET 5). The default for .NET 8 and above is bookworm, running Debian 12. This is what the base containers should look like.

> mcr.microsoft.com/dotnet/aspnet:8.0-bookworm-slim
> mcr.microsoft.com/dotnet/sdk:8.0-bookworm
